application/src/Api/Adapter/AbstractEntityAdapter.php

```diff
@@ -235,14 +235,21 @@ public function search(Request $request)
 
         $scalarField = $request->getOption('returnScalar');
         if ($scalarField) {
-            $fieldNames = $this->getEntityManager()->getClassMetadata($entityClass)->getFieldNames();
+            $classMetadata = $this->getEntityManager()->getClassMetadata($entityClass);
+            $fieldNames = $classMetadata->getFieldNames();
             if (!in_array($scalarField, $fieldNames)) {
-                throw new Exception\BadRequestException(sprintf(
-                    $this->getTranslator()->translate('The "%1$s" field is not available in the %2$s entity class.'),
-                    $scalarField, $entityClass
-                ));
+                $associationNames = $classMetadata->getAssociationNames();
+                if (!in_array($scalarField, $associationNames)) {
+                    throw new Exception\BadRequestException(sprintf(
+                        $this->getTranslator()->translate('The "%1$s" field is not available in the %2$s entity class.'),
+                        $scalarField, $entityClass
+                    ));
+                }
+                $qb->join('omeka_root.' . $scalarField, $scalarField . '_');
```

Comment:
I think it's better off using `createAlias()` for this join's alias as is usual for other joins.